File: same_day_delivery_client/lib/model/product.model.dart
import 'dart:convert';

class ProductModel {
  String productId;
  String sellerId;
  String productName;
  num productPrice;
  String productImage;
  String productDescription;
  String productDate;
  String productLocation;

  ProductModel({
    required this.productId,
    required this.sellerId,
    required this.productName,
    required this.productPrice,
    required this.productImage,
    required this.productDescription,
    required this.productDate,
    required this.productLocation,
  });

  Map<String, dynamic> toJson() {
    return {
      '_id': productId,
      'seller_id': sellerId,
      'product_name': productName,
      'product_price': productPrice,
      'product_image': productImage,
      'product_description': productDescription,
      'product_date': productDate,
      'product_location': productLocation,
    };
  }

  factory ProductModel.fromJson(Map<String, dynamic> json) {
    return ProductModel(
      productId: json['_id'],
      sellerId: json['seller_id'],
      productName: json['product_name'],
      productPrice: json['product_price'],
      productImage: json['product_image'],
      productDescription: json['product_description'],
      productDate: json['product_date'],
      productLocation: json['product_location'] ?? "",
    );
  }

  String toJsonString() {
    return ("""{
      "productID": "$productId",
      "sellerID": "$sellerId",
      "productName": "$productName",
      "productPrice": $productPrice,
      "productImage": "$productImage",
      "productDescription": "$productDescription",
      "productDate": "$productDate",
      "productLocation": "$productLocation"
    }""");
  }

  factory ProductModel.fromJsonString(String jsonString) {
    Map<String, dynamic> json = jsonDecode(jsonString);
    return ProductModel(
      productId: json['productID'],
      sellerId: json['sellerID'],
      productName: json['productName'],
      productPrice: json['productPrice'],
      productImage: json['productImage'],
      productDescription: json['productDescription'],
      productDate: json['productDate'],
      productLocation: json['productLocation'] ?? "",
    );
  }
}

File: SameDayDelivery-Admin/model/product.model.js
const mongoose = require('mongoose');

const productModel = new mongoose.Schema({
    product_id: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        default:new mongoose.Types.ObjectId,
    },
    seller_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Seller",
        required: true,
    },
    product_name: {
        type: String,
        required: true,
        index: true,
    },
    product_price: {
        type: Number,
        required: true,
    },
    product_image: {
        type: String,
        required: true,
    },
    product_description: {
        type: String,
        required: true,
    },
    product_date: {
        type: Date,
        required: true,
        default: Date.now,
    },
    product_category:{
        type:String,
        required:true,
        enum:["food","clothing","electronics","furniture","medicine","other"],
        default:"other",
    }
});

module.exports = mongoose.model("Product", productModel);